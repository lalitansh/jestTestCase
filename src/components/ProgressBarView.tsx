import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {Dialog} from 'react-native-simple-dialogs';
import style from '../constants/theme/Style';
import {color} from '../constants/theme/Color';
import {MaterialIndicator} from 'react-native-indicators';

type Props = {
  visible: boolean;
  loadingText?: string;
};

type State = {};

export default class ProgressBarView extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
  }

  render() {
    const {loadingText, visible} = this.props || {};
    if (visible) {
      return (
        <Dialog dialogStyle={style.progressDialogBox} visible={true}>
          <View style={styles.viewStyle}>
            <MaterialIndicator
              color={color.primary}
              size={40}
              animationDuration={5200}
              trackWidth={3}
            />

            {loadingText && (
              <Text style={{marginTop: 5, textAlign: 'center'}}>
                {loadingText}
              </Text>
            )}
          </View>
        </Dialog>
      );
    } else {
      return <View />;
    }
  }
}

const styles = StyleSheet.create({
  viewStyle: {
    height: 40,
    alignItem: 'center',
  },
});
