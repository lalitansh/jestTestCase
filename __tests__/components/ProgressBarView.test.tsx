import renderer from 'react-test-renderer';
import React from 'react';
import ProgressBarView from '../../src/components/ProgressBarView';

const mockVisible = {
  visible: false,
};

it('renders correctly progress bar', () => {
  const tree = renderer.create(<ProgressBarView {...mockVisible} />).toJSON();
  expect(tree).toMatchSnapshot();
});
