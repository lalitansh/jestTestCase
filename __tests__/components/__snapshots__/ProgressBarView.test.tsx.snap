// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly progress bar 1`] = `<View />`;
